src/lib/operators/join_hash.cpp

```diff
@@ -196,12 +200,12 @@ constexpr Radix hash_value(const OriginalType& value, const unsigned int seed) {
 
 template <typename T, typename HashedType>
 std::shared_ptr<Partition<T>> materialize_input(const std::shared_ptr<const Table>& in_table, ColumnID column_id,
-                                                std::vector<std::vector<size_t>>& histograms,
-                                                const size_t radix_bits, const unsigned int partitioning_seed,
-                                                bool keep_nulls = false) {
+                                                std::vector<std::vector<size_t>>& histograms, const size_t radix_bits,
+                                                const unsigned int partitioning_seed, bool keep_nulls = false) {
   // list of all elements that will be partitioned
-  auto elements = std::make_shared<Partition<T>>();
-  elements->resize(in_table->row_count());
+  auto elements = std::make_shared<Partition<T>>(in_table->row_count());
+
+  for (auto& x : *elements) x = PartitionedElement<T>();
```

Comment:
I'm a bit confused about this line. Why do we use the uninitialized vector if we are going to initialize it right after anyway?

Reply:
D'oh. That's left-over from debugging. :/